### DISCUSSION TITLE
Install using LazyVim

### DISCUSSION THREAD

**Discussion by kellybrower** [mrcjkb/haskell-tools.nvim#314] [General]
Hello.  I am simply asking for instructions to correctly install using LazyVim.  I am quite new to NeoVim.  I know that the community might not like to see questions like this, so sorry in advance.  I just simply don't understand where to go from the simple instructions that just include 

```
{
  'mrcjkb/haskell-tools.nvim',
  dependencies = {
    'nvim-lua/plenary.nvim',
  },
  version = '^2', -- Recommended
  ft = { 'haskell', 'lhaskell', 'cabal', 'cabalproject' },
}

```

Thanks in advance for your help.

**Comment by kellybrower** [comment]
What I've done is saved in plugins

```
  return {
  "mrcjkb/haskell-tools.nvim",
  dependencies = {
    "nvim-lua/plenary.nvim",
  },
  version = "^2", -- Recommended
  ft = { "haskell", "lhaskell", "cabal", "cabalproject" },
}
```

But I get an installation error saying:

```
    ○ haskell-tools.nvim  haskell  cabalproject  cabal  lhaskell 
        ...al/share/nvim/lazy/lazy.nvim/lua/lazy/manage/task/fs.lua:16: attempt to index local 'stat' (a nil value)

**Reply by kellybrower** [reply]
with a full message below
```    ○ haskell-tools.nvim  haskell  cabalproject  cabal  lhaskell 
        dir /Users/kellybrower/.local/share/nvim/lazy/haskell-tools.nvim
        url https://github.com/mrcjkb/haskell-tools.nvim
        ft   haskell  cabalproject  cabal  lhaskell 

      ✔ [task] clean 0.13ms
        ...al/share/nvim/lazy/lazy.nvim/lua/lazy/manage/task/fs.lua:16: attempt to index local 'stat' (a nil value)
```

**Reply by mrcjkb** [reply]
Hey :wave: 

No need to apologise for asking questions :wink: 
Regarding the use with LazyVim, someone [recently requested a Haskell language pack](https://github.com/mrcjkb/haskell-tools.nvim/issues/290) with this plugin, which resulted in [this PR](https://github.com/LazyVim/LazyVim/pull/2052).

I don't use LazyVim and don't have the capacity to maintain the language pack, but it has been tested and confirmed working.
So you could use that as a starting point.

Regarding the error message `...al/share/nvim/lazy/lazy.nvim/lua/lazy/manage/task/fs.lua:16: attempt to index local 'stat' (a nil value)`
That's specific to [lazy.nvim](https://github.com/folke/lazy.nvim) (the plugin manager that LazyVim uses). It looks like there's a missing `nil` check (Lua's equivalent to `null`) in the lazy.nvim code base.

Judging by the fact that it occurs in `[task] clean`, I would guess that there's a bug in which it deletes a directory and then tries to access it?
